packages/diagram/src/module/diagrams/content/uml/classifier/ports.ts

```diff
@@ -0,0 +1,105 @@
+                    {
+                        value: fun([
+                            ...parse(
+                                `
+                                    this.scope = args.scope
+                                    this.canvasScope = args.canvasScope
+                                    this.element = args.element
+                                    scope.ports = list()
+                                `
+                            ),
+                            id("scope").assignField(
+                                "port",
+                                fun(
+                                    `
+                                        (pos, optionalCallback) = args
+                                        callback = optionalCallback ?? {}
+                                        portElement = canvasElement(
+                                            class = list("port-element"),
+                                            content = rect(class = list("port")),
+                                            pos = canvasScope.lpos(element, pos, args.dist ?? -1)
+                                        )
+                                        
+                                        result = []
+                                        scope.internal.providesRequiresContentHandler[0](
+                                            scope = result,
+                                            args = args,
+                                            element = portElement,
+                                            canvasScope = canvasScope
+                                        )
+                                        callback.callWithScope(result)
+                                        
+                                        scope.internal.registerCanvasElement(portElement, args, canvasScope)
+                                    `,
+                                    {
+                                        docs: "Creates a port on the outline of the classifier",
+                                        params: [
+                                            [
+                                                0,
+                                                "Relative position on the outline",
+                                                optional(LinePointLayoutConfig.POS_TYPE)
+                                            ],
+                                            [
+                                                1,
+                                                "Optional callback function to define the content of the port",
+                                                optional(functionType)
+                                            ],
+                                            [
+                                                "dist",
+                                                "Distance from the outline, needs to be adapted if width of outline is not the default value",
```

Comment:
```suggestion
                                                "Distance from the outline, defaults to the width of the outline",
```

Reply:
No it does not
defaults to half of the width of the outline

Reply:
Changed

Reply:
changed to sth similar with 1dbd203d8081c0a256cbc232443c4150ce611fdf